### ISSUE TITLE
Different preprocessing for train, test and val sets

### ISSUE THREAD

**Issue by samkohn** [NERSC/dayabay-learn#11]
The way the code is set up currently, the mean and std are calculated separately for all 3 data sets. This means that the transformations are slightly different. E.g. the mean for the train set is slightly different from the mean for the test set, hence they will be transformed differently when they are centered on 0.

Is this something we want? I think it makes more sense to save the transformation used on the training set, and apply it directly to the other sets without regard to the properties of the input.

If you agree with me, I propose the following interface to manage the data preprocessing:

A preprocess_data function would accept as a parameter the train set and preprocess it in-place. It would return a new function encapsulating the transformation it figured out on the train set. This function could then be called on the val and test sets to apply the same transformation.

This is similar to the behavior of [scikit-learn's StandardScaler](http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html), which has a "fit" method to determine the preprocessing parameters, and a transform method to apply them repeatedly to different data sets.
